src/main/java/org/broadinstitute/hellbender/utils/diffengine/DiffNode.java

```diff
@@ -0,0 +1,181 @@
+package org.broadinstitute.hellbender.utils.diffengine;
+
+import com.google.common.base.Strings;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+
+/**
+ * An interface that must be implemented to allow us to calculate differences
+ * between structured objects
+ */
+final class DiffNode extends DiffValue {
+
+    private DiffNode(final Map<String, DiffElement> elements) {
+        super(elements);
+    }
+
+    private DiffNode(final DiffElement binding, final Map<String, DiffElement> elements) {
+        super(binding, elements);
+    }
+
+    public static DiffNode rooted(final String name) {
+        return empty(name, DiffElement.ROOT);
+    }
+
+    public static DiffNode empty(final String name, final DiffElement parent) {
+        final DiffNode df = new DiffNode(new HashMap<>());
+        final DiffElement elt = new DiffElement(name, parent, df);
+        df.setBinding(elt);
+        return df;
+    }
+
+    @Override
+    public Map<String, DiffElement> getValue() {
+        @SuppressWarnings("unchecked")
+        final Map<String, DiffElement> v = (Map<String, DiffElement>)super.getValue();
```

Comment:
There is some questionable design going on here with the superclass storing the value in an untyped `Object` field...